### ISSUE TITLE
cursor disappearing

### ISSUE THREAD

**Issue by stephanie-ice** [microsoft/vscode#84299]
Issue Type: <b>Bug</b>

My cursor keeps randomly disappearing.  I'm on a brand new Mac Mojave version 10.14.6, and I was running Chrome, VSCode, and Terminal.  I quit all the applications and ran one app at a time; I have both Terminal and VSCode open now, and the cursor randomly disappears every few minutes or even immediately. I must close my laptop and reopen it to get the cursor back.

VS Code version: Code 1.40.0 (86405ea23e3937316009fc27c9361deee66ffbf5, 2019-11-06T17:09:34.601Z)
OS version: Darwin x64 18.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz (8 x 1400)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|8.00GB (2.27GB free)|
|Process Argv|.|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
open-in-browser|tec|2.0.0

</details>

**Comment by alexandrudima** [comment]
@stephanie-ice What do you mean the cursor disappears? Do you mean the mouse cursor or the text insertion cursor (that blinks in the editor)?

**Comment by antoinell** [comment]
I also see my cursor disappearing but I am not using vscode. I am using citrix workspace to connnect to a corporate Windows PC. Moving the mouse pointer towards the top center of the screen makes the mouse pointer come back. I guess where my mouse pointer is because controls such as buttons inside web pages change aspect when the mouse pointer is on top of them.

**Comment by alexandrudima** [comment]
I think this might be an issue with the remote desktoping software.

**Comment by yoctozepto** [comment]
Since 1.40.0 vscode integrated terminal cursor started disappearing for me on CentOS 7.
The 100% way to reproduce is to run vim and move cursor down or up a line. It reappears when moving it back and forward (left/right). This does not happen with 1.39 so it's something that changed in vscode (or its deps more likely).
This does not affect editor cursor, only terminal.

**Comment by Tyriar** [comment]
Please provide a screenshot illustrating the problem. See [this wiki page](https://github.com/microsoft/vscode/wiki/Terminal-Issues) for more info on how to report actionable terminal issues.

**Comment by yoctozepto** [comment]
![image](https://user-images.githubusercontent.com/3455956/69260711-99316900-0bc0-11ea-8dbc-475491cd8391.png)

Create a more-than-one-line file and open it with vim. Move cursor down with either j or down arrow. The cursor is gone. Move it left or right and it is back. Only changing the line seems to trigger this bug.

**Comment by Tyriar** [comment]
What theme are you using? Does the problem go away if you change themes?

**Comment by yoctozepto** [comment]
Light+, changing theme does not amend it...

**Comment by Tyriar** [comment]
Ca you share your settings.json file?

**Comment by monkeygroover** [comment]
Hi, seeing this too, cursor disappears when I press up within vim running in the integrated terminal, also this is a recent issue, I don't remember seeing it before the last update.

Things I've ruled out so far, doesn't appear to be;

1) OS (happening for me on both OSX and Ubuntu 18.04)
2) Shell (happens when the terminal is running both bash and fish)
3) Theme (happens in all themes i've tried)
4) Font (tried a few, same issue with each)

I'll try disabling extensions and attach a settings.json when I get a chance.

thank you.

**Comment by yoctozepto** [comment]
@Tyriar bingo!

This is the culprit:
    "terminal.integrated.rendererType": "dom",
I forgot I had this. I think vscode recommended this to me and I accepted. 🤷

PS: The font got crisper as well when I removed that. No problems so far.

**Comment by monkeygroover** [comment]
@yoctozepto yep! can confirm that fixes it for me too. Cheers.

**Comment by rlipscombe** [comment]
I've got the same problem on my Windows PC that has rendererType=dom. Cursor disappears if the window doesn't get redrawn (e.g. vi doesn't need to scroll the view).

Steps:
1. Open VS Code (on Windows for me) with rendererType=dom; I had it pointing to a WSL2-hosted Rust project.
2. Open the terminal. It should be 6 lines tall.
3. Because I was using WSL2, it ran bash. Then run "vi Cargo.toml".
4. Use the cursor to move down.
5. Use the cursor to move up. Note that the cursor disappears between line 3 and line 2 (as in: when the screen doesn't need to scroll, but the cursor needs to move). It reappears when you move up to line 1 (in my case, probably because I've got matching-paren highlighting turned on in vim, causing a redraw).

Setting the renderer back to "auto" fixes the problem.
